browse/routes/ui.py

```diff
@@ -64,97 +83,201 @@ def abstract(arxiv_id: str) -> Response:
+                 methods=['GET'],
+                 defaults={'trackback_id': '', 'hashed_document_id': ''})
+@blueprint.route('tb/redirect/<string:trackback_id>/<string:hashed_document_id>',
+                 methods=['GET'])
+def tb_redirect(trackback_id: str, hashed_document_id: str) -> Response:
+    """Get the trackback redirect link."""
+    response, code, headers = tb_page.get_tb_redirect(trackback_id,
+                                                      hashed_document_id)
+    if code == status.HTTP_301_MOVED_PERMANENTLY:
+        return redirect(headers['Location'], code=code)  # type: ignore
+    raise InternalServerError('Unexpected error')
+
+
+@blueprint.route('prevnext', methods=['GET', 'POST'])
+def previous_next() -> Union[str, Response]:
+    """Previous/Next navigation used on /abs page."""
+    if not request.args:
+        raise BadRequest
+    response, code, headers = prevnext.get_prevnext(request.args)
+    if code == status.HTTP_301_MOVED_PERMANENTLY:
+        return redirect(headers['Location'], code=code)  # type: ignore
+    raise InternalServerError('Unexpected error')
+
+
+@blueprint.route('trackback/', methods=['GET'], defaults={'arxiv_id': ''})
+@blueprint.route('trackback/<path:arxiv_id>', methods=['GET', 'POST'])
 def trackback(arxiv_id: str) -> Union[str, Response]:
     """Route to define new trackbacks for papers."""
     raise InternalServerError(f'Not Yet Implemented {arxiv_id}')
 
 
-@blueprint.route('/ct')
+@blueprint.route('ct')
 def clickthrough() -> Response:
     """Generate redirect for clickthrough links."""
     if 'url' in request.args and 'v' in request.args:
         if is_hash_valid(current_app.config['CLICKTHROUGH_SECRET'],
                          request.args.get('url'),
                          request.args.get('v')):
-            return redirect(request.args.get('url'))
+            return redirect(request.args.get('url'))  # type: ignore
         else:
             raise BadRequest('Bad click-through redirect')
 
     raise NotFound
 
 
-@blueprint.route('/list/<context>/<subcontext>')
-def list_articles(current_context: str, yymm: str) -> Response:
-    """
-    List articles by context, month etc.
+@blueprint.route('list', defaults={'context': '', 'subcontext': ''},
+                 methods=['GET', 'POST'])
+@blueprint.route('list/', defaults={'context': '', 'subcontext': ''},
+                 methods=['GET', 'POST'])
+@blueprint.route('list/<context>/<subcontext>', methods=['GET', 'POST'])
+def list_articles(context: str, subcontext: str) -> Response:
+    """List articles by context, month etc.
 
-    Context might be a context or an archive
-    Subcontext should be 'recent' 'new' or a string of format yymm
+    Context might be a context or an archive Subcontext should be
+    'recent' 'new' or a string of format yymm
     """
-    raise InternalServerError(f'Not yet implemented {current_context} {yymm}')
+    if request.args.get('archive', None) is not None:
```

Comment:
L175-L181 are looking a lot like controller logic. Can we move this to ``list_page.get_listing()``?